skimage/io/tests/test_pil.py

```diff
@@ -205,6 +205,21 @@ def test_imsave_filelike():
     assert out.shape == shape
     assert_allclose(out, image)
 
+def test_imsave_boolean_input():
+    shape = (2, 2)
+    image = np.zeros(shape)
+    s = BytesIO()
```

Comment:
You can use tempfile-decorator here (see https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile).